src/05-derivatives/05-diffusion-derivatives.md

```diff
@@ -18,450 +18,700 @@ should be included in the pipeline documentation.
-    -   `NJumps` : value
-    -   `SampleEvery` : value
-    -   `Model` : {`monoexponential`,`multiexponentialStick`,`multiexponentialZeppelin`}
+-  OPTIONAL files "`<source_keywords>[_space-<space>][_desc-<label>]_parameter-<extparam*>_<model>.json`"
+   may provide only information or parameters relevant to derivation of each
+   relevant [extrinsic](#paramdef-extrinsic) model parameter image.
 
-Examples:
+-  If all [intrinsic](#paramdef-intrinsic) model parameters are
+   contained within a single image file, field "`_parameter-*`" MUST be
+   omitted (for intrinsic parameter image only):
 
-A Diffusion Tensor fit:
+   ```Text
+   <pipeline_name>/
+       sub-<participant_label>/
+           dwi/
+               <source_keywords>[_space-<space>][_desc-<label>]_<model>.nii[.gz]
+               <source_keywords>[_space-<space>][_desc-<label>]_<model>.json
+   ```
 
-```Text
-my_diffusion_pipeline/
-    sub-01/
-        dwi/
-            sub-01_space-T1w_desc-WLS_model-DTI_diffmodel.nii.gz
-            sub-01_space-T1w_desc-WLS_model-DTI_diffmodel.json
-```
+### Data representations
 
-Contents of JSON file:
+There are multiple techniques by which data that relate to the anisotropic
+nature of either the diffusion process or underlying tissue may be
+arranged and/or encoded into NIfTI image data. A list of known techniques
+is enumerated below, accompanied by requisite information specific to the
+reading / writing of each representation.
 
-```JSON
-{
-    "Parameters": {
-        "FitMethod": "WLS"
-    }
-}
-```
+1. <a name="data-scalar">*Scalars*</a>:
 
-A multi-shell, multi-tissue Constrained Spherical Deconvolution fit:
+   Any model parameter image (whether [intrinsic](#paramdef-intrinsic) or
+   [extrinsic](#paramdef-extrinsic)) where a solitary numerical value is
+   defined in each 3D image voxel is referred to here as a "scalar" image.
 
-```Text
-my_diffusion_pipeline/
-    sub-01/
-        dwi/
-            sub-01_model-CSD_desc-WM_diffmodel.nii.gz
-            sub-01_model-CSD_desc-WM_diffmodel.json
-            sub-01_model-CSD_desc-GM_diffmodel.nii.gz
-            sub-01_model-CSD_desc-GM_diffmodel.json
-            sub-01_model-CSD_desc-CSF_diffmodel.nii.gz
-            sub-01_model-CSD_desc-CSF_diffmodel.json
-```
+1. <a name="data-dec">*Directionally-Encoded Colours (DEC)*</a>:
 
-Contents of JSON file (using "`_tissue-WM`" as example):
-
-```JSON
-{
-    "Shells": [ 0, 1000, 2000, 3000 ],
-    "Parameters": {
-        "Tissue": "White matter",
-        "SphericalHarmonicDegree": 8,
-        "ResponseFunctionZSH": [ [ 15335 0 0 0 0 0 ],
-                                 [ 7688 -2766 597 -82 10 -5 ],
-                                 [ 5258 -2592 1024 -270 48 -11 ],
-                                 [ 4226 -2193 1124 -436 119 -19 ] ]
-        "SphericalHarmonicBasis": "MRtrix3",
-        "NonNegativityConstraint": "hard"
-    }
-}
-```
+   4D image with three volumes, intended to be interpreted as red, green

```

Comment:
Here, I know that we had some discussions on this during the initial derivatives meeting. This format is what most tools now use, but is not the official NIfTI way (using a special datatype and intent code). I think we should probably support both, since, for example, Trackvis still expects the official NIfTI standard.

Reply:
This was raised during the meeting; I was not aware of this format previously. Part of the issue is that many softwares ignore NIfTI intent fields. It would theoretically be possible to support both; but it would I think at least require identification of use of this format within the JSON sidecar rather than relying on the NIfTI intent code.

Reply:
I agree that most software just disregard intent codes. I think identifying it in the sidecar is a good idea, but I would still allow using the intent code for tools that support it natively.